packages/react/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,271 @@
+import {
+  childrenExist,
+  commonPropTypes,
+  createShorthandFactory,
+  UIComponentProps,
+  ChildrenComponentProps,
+  rtlTextContainer,
+  applyAccessibilityKeyHandlers,
+  UIComponent,
+} from '../../lib'
+import {
+  ShorthandRenderFunction,
+  WithAsProp,
+  withSafeTypeForAs,
+  ShorthandCollection,
+  ShorthandValue,
+} from '../../types'
+import { Accessibility } from '../../lib/accessibility/types'
+import { treeBehavior } from '../../lib/accessibility'
+import { getFirstFocusable } from '../../lib/accessibility/FocusZone/focusUtilities'
+
+export interface TreeSlotClassNames {
+  item: string
+}
+
+export interface TreeProps extends UIComponentProps, ChildrenComponentProps {
+  /** Accessibility behavior if overridden by the user. */
+  accessibility?: Accessibility
+
+  /** Initial activeIndex value. */
+  defaultActiveItemsList?: ShorthandCollection<TreeItemProps>
+
+  /** Only allow one subtree to be open at a time. */
+  exclusive?: boolean
+
+  /** Shorthand array of props for Tree. */
+  items?: ShorthandCollection<TreeItemProps>
+
+  /**
+   * A custom render function for the title slot.
+   *
+   * @param {React.ReactType} Component - The computed component for this slot.
+   * @param {object} props - The computed props for this slot.
+   * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+   */
+  renderItemTitle?: ShorthandRenderFunction
+}
+
+export interface TreeState {
+  activeItems: Map<ShorthandValue<TreeItemProps>, { open: boolean; element: HTMLElement }>
+}
+
+class Tree extends UIComponent<WithAsProp<TreeProps>, TreeState> {
+  static create: Function
+
+  static displayName = 'Tree'
+
+  static className = 'ui-tree'
+
+  static slotClassNames: TreeSlotClassNames = {
+    item: `${Tree.className}__item`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    activeItemsList: customPropTypes.every([
+      customPropTypes.disallow(['children']),
+      PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.number), PropTypes.number]),
+    ]),
+    defaultActiveItemsList: customPropTypes.every([
+      customPropTypes.disallow(['children']),
+      PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.number), PropTypes.number]),
+    ]),
+    exclusive: PropTypes.bool,
+    items: customPropTypes.collectionShorthand,
+    renderItemTitle: PropTypes.func,
+    rtlAttributes: PropTypes.func,
+    onactiveItemsListChange: PropTypes.func,
+  }
+
+  static defaultProps = {
+    as: 'div',
+    accessibility: treeBehavior,
+  }
+
+  static autoControlledProps = ['activeItemsList']
+
+  itemRefs = []
+
+  constructor(props, context) {
+    super(props, context)
+
+    const activeItems = new Map()
+
+    if (props.items) {
+      const setItemsLevelAndSize = (items = props.items, level = 1, parent?) => {
+        items.forEach((item: ShorthandValue<TreeItemProps>, index: number) => {
+          item['level'] = level
```

Comment:
This mutates the original `props.items` and it's definitely unexpected because `props` should be immutable:

![image](https://user-images.githubusercontent.com/14183168/63342265-535bcf00-c34b-11e9-9fed-bd849403c514.png)